PortalSwift/Sources/PortalSwift/Core/Api/PortalApi.swift

```diff
@@ -0,0 +1,589 @@
+//
+//  PortalApi.swift
+//
+//  Created by Portal Labs, Inc.
+//  Copyright © 2022 Portal Labs, Inc. All rights reserved.
+//
+
+import Foundation
+
+/// The class to interface with Portal's REST API.
+public class PortalApi {
+  private var _client: ClientResponse?
+  private var apiKey: String
+  private var baseUrl: String
+  private let decoder = JSONDecoder()
+  private var httpRequests: HttpRequester
+  private let logger = PortalLogger()
+  private var provider: PortalProvider?
+  private let requests: PortalRequests
+  private let featureFlags: FeatureFlags?
+
+  private var address: String? {
+    self.provider?.address
```

Comment:
Just confirming, but it does makes sense to just return nil here when `self.provider` is nil (like you're currently doing) instead of throwing an error, right?

Reply:
If we get this far and the Provider hasn't initialized properly, we should be throwing an error (because it means the error wasn't thrown during initialization).